ndc-models/src/lib.rs

```diff
@@ -330,10 +334,22 @@ pub enum ComparisonOperatorDefinition {
 // ANCHOR: AggregateFunctionDefinition
 /// The definition of an aggregation function on a scalar type
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(tag = "type", rename_all = "snake_case")]
 #[schemars(title = "Aggregate Function Definition")]
-pub struct AggregateFunctionDefinition {
-    /// The scalar or object type of the result of this function
-    pub result_type: Type,
+pub enum AggregateFunctionDefinition {
+    Min,
+    Max,
+    Sum {
+        /// The scalar type of the result of this function, which should have
+        /// one of the type representations Int64 or Float64, depending on
```

Comment:
This isn't always the case (Int64 or Float64). For example, the postgres connector's Float4 (ie 32-bit float) sum function returns Float4. However, all integer types do appear to sum to a 64-bit int.

Reply:
Right, but a connector can always upcast a smaller int or float type to a 64-bit type, and since we can't control how many rows will be summed, we should pick the largest possible representations.

Reply:
I suppose I'll add that MongoDB has a Float128 type which I would want to use with inputs of that type

Reply:
> Right, but a connector can always upcast a smaller int or float type to a 64-bit type, and since we can't control how many rows will be summed, we should pick the largest possible representations.

Makes sense. [Here](https://substrait.io/extensions/functions_arithmetic/#sum) is how `sum is defined in substrait.

Reply:
This means that, in something like Postgres, we'd have to cast up any Float4 column to Float8 before aggregation in order to meet this spec. This could be seen as not respecting the choice of the end-user when they chose Float4 precision for their column. Postgres seems to deliberately retain that precision during aggregation (which it doesn't do for integers, only floats, from what I can see in our PG connector schema), probably because you can sum up pretty far using floats, you just lose precision, unlike integers where you overflow. 

Maybe overriding the chosen precision is okay, but it's something to consider.

Reply:
For 128-bit numeric types, unfortunately those wouldn't be supported by datafusion right now. We could still support custom aggregate operators for those types, but just not the standardized sum operator.